gokart/mypy.py

```diff
@@ -329,6 +330,10 @@ def collect_attributes(self) -> Optional[list[TaskOnKartAttribute]]:
             with state.strict_optional_set(self._api.options.strict_optional):
                 init_type = self._infer_task_on_kart_attr_init_type(sym, stmt)
```

Comment:
[must]
currently `init_type` is inferred from `default` value.

But `_generate_type_from_parameters` should be prioritize to `default` type.

In addition, should we compare the type of default and inferred type from parameter class, though I do not know the way.